tests/ttnn/unit_tests/operations/test_uniform.py

```diff
@@ -0,0 +1,109 @@
+# SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+
+# SPDX-License-Identifier: Apache-2.0
+
+import torch
+import pytest
+import numpy as np
+import ttnn
+from loguru import logger
+from tests.ttnn.unit_tests.operations.test_utils import (
+    get_compute_kernel_options,
+    compute_kernel_options,
+    compute_kernel_ids,
+    to_cpu,
+    to_npu,
+)
+
+
+def get_lib_dtype(lib, dtype):
+    """Maps dtype to corresponding library dtype."""
+    dtype_map = {
+        "bfloat16": lib.bfloat16,
+        "float32": lib.float32,
+    }
+    return dtype_map.get(dtype, None)
+
+
+def run_uniform(shape, rand_range, dtype, device, compute_kernel_options=None):
+    compute_kernel_config = get_compute_kernel_options(compute_kernel_options)
+    rand_from, rand_to = rand_range[0], rand_range[1]
+    cpu_input = torch.ones(shape, dtype=get_lib_dtype(torch, dtype))
+    npu_input = to_npu(cpu_input, device, npu_dtype=get_lib_dtype(ttnn, dtype))
+
+    # Exec uniform operation on ttnn using NPU
+    ttnn.uniform(npu_input, rand_from, rand_to, compute_kernel_config=compute_kernel_config)
+    tt_input = to_cpu(npu_input, shape)
+
+    hash_mp = {}
+    for element in tt_input.flatten():
+        hash_mp[element.item()] = hash_mp.get(element.item(), 0) + 1
+
+    expected_mean, expected_var = (rand_from + rand_to) / 2, pow(rand_to - rand_from, 2) / 12
+    npu_mean, npu_var = torch.mean(tt_input).item(), torch.var(tt_input).item()
+    min_val, max_val = torch.min(tt_input).item(), torch.max(tt_input).item()
+
+    logger.info(f"Distinct elements: {len(list(hash_mp.keys()))}")
+    logger.info(f"Expected mean: {expected_mean}, NPU mean: {npu_mean}")
+    logger.info(f"Expected var: {expected_var}, NPU var: {npu_var}")
+
+    assert rand_from <= min_val and max_val <= rand_to
```

Comment:
```suggestion
    assert rand_from <= min_val and max_val < rand_to
```
I understand that uniform does not include the rand_to value.

Reply:
Same as this cmt: https://github.com/tenstorrent/tt-metal/pull/13735#discussion_r1796747564

Reply:
Updated in this [commit](https://github.com/tenstorrent/tt-metal/pull/13735/commits/331cd75c78cba9765c567c53a9fcdd9921b0e9d0).